copy/copy.go

```diff
@@ -771,6 +948,8 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, srcInfo types.BlobInfo,
 	getOriginalLayerCopyWriter func(decompressor compression.DecompressorFunc) io.Writer,
 	canModifyBlob bool, isConfig bool, bar *mpb.Bar) (types.BlobInfo, error) {
+	c.semMaxParallelCopies.Acquire(ctx, 1)
```

Comment:
Moving the enforcement of `copyInParallel` all the way down here does not work:
- `GetBlob` has already been called, in parallel, even if `!HasThreadSafeGetBlob`
- Similarly, `TryReusingBlob` has been called, in parallel; that’s usually not safe. (Yes, `HasThreadSafePutBlob` is documented incorrectly.)

Sure, it could be a good idea to allow more goroutines and fewer actual parallel data transfers, _if_ parallelism is supported; but that should _both_ be conditional on that condition, and either way it’s quite independent from the purpose of this PR AFAICS.

Reply:
I'll delete the commits from the PR.